### PR TITLE
Add a leadtime argument to the chgres_cube driver.

### LINKED ISSUE #640: The `chgres_cube` driver needs a `--leadtime` argument.
If this is a bug, please consider using the [Bug Report](./bug_report.md) template.
Your issue may already have been created!

Please look through the existing issues on the [Issue tracker](https://github.com/ufs-community/uwtools/issues) to see if this issue has been created.
If so, please consider using that Issue to add any additional information.

**Description**

When running the `chgres_cube` driver to create lateral boundary conditions, the namelist expects the date information to be consistent with the valid-time of the forecast file being processed, however, these forecast times are prepared for a given `cycle`, so we will need to pass both pieces of information to get the expected behavior -- the `rundir` has `cycle` information, but the namelist has `cycle + leadtime` information.

**Requirements**

**Describe alternatives you've considered**

This could be handled as an environment variable, but would be horribly inconsistent with the other drivers that do have an optional lead time argument.

**Acceptance Criteria (Definition of Done)**

- [ ] The chgres_cube driver can be appropriately configured for both initial and lateral boundary condition tasks with output behavior similar to that in existing apps.
- [ ] Docs and tests updated to support the new argument.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Synopsis**

Fixes #640 

**Type**

- [x] Bug fix (corrects a known issue)
- [ ] Code maintenance (refactoring, etc. without behavior change)
- [ ] Documentation
- [x] Enhancement (adds new functionality)
- [ ] Tooling (CI, code-quality, packaging, revision-control, etc.)

**Impact**

- [ ] This is a breaking change (changes existing functionality)
- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

**Review by elcarpenterNOAA — APPROVED** [review]

**Review by WeirAE — APPROVED** [review]
Looks clean and should help to smooth out the SRW code too, LGTM

**Review by nbharwani-cu — APPROVED** [review]

File: src/uwtools/tests/drivers/test_chgres_cube.py [hunk 1]
```diff
@@ -165,4 +170,4 @@ def test_ChgresCube_runscript(driverobj):
 
 
 def test_ChgresCube_taskname(driverobj):
-    assert driverobj.taskname("foo") == "20240201 18Z chgres_cube foo"
+    assert driverobj.taskname("foo") == "20240202 18:00:00 chgres_cube foo"
```
Comment (maddenp-noaa) [inline src/uwtools/tests/drivers/test_chgres_cube.py:173]: I know that this output is from existing behavior, but seeing it now I wonder if it's what we really want for a cycle + leadtime driver. This gives a valid time for the data, but the connection to the cycle is lost. I mean, alternatively, we could have `20240201 18Z 24:00:00`, i.e. this is the Feb 1 18Z cycle, with a 24-hour leadtime. On the other hand it _does_ seem handy to not have to calculate the valid time -- like what time does the hour-97 forecast of some cycle actually correspond to? I can see it both ways, and am not asking for a change, just musing.

**Review by maddenp-cu — APPROVED** [review]

**Review by Byrnetp — APPROVED** [review]

File: src/uwtools/tests/drivers/test_chgres_cube.py (hunk 1 above)
Comment (christinaholtNOAA) [reply src/uwtools/tests/drivers/test_chgres_cube.py:173]: I could see reporting `20240201 18Z 24:00:00` as an improvement. We typically don't care about valid time in these logs. I think the only tasks that really care about valid time for user-facing aspects is verification.
